aten/src/ATen/native/transformers/cuda/attention_backward.cu

```diff
@@ -68,6 +71,68 @@ namespace at {
 
 namespace native {
 
+std::tuple<Tensor, Tensor, Tensor> _flash_attention_backward(
+    const Tensor& grad_out,
+    const Tensor& query,
+    const Tensor& key,
+    const Tensor& value,
+    const Tensor& out,
+    const Tensor& logsumexp,
+    const Tensor& cumulative_sequence_length_q,
+    const Tensor& cumulative_sequence_length_k,
+    const int64_t max_seqlen_batch_q,
+    const int64_t max_seqlen_batch_k,
+    double dropout_p,
+    bool is_causal,
+    const int64_t philox_seed,
+    const int64_t philox_offset) {
+#if defined(USE_FLASH_ATTENTION)
+  /*
+  num_splits determines how much to parallelize over the seqlen_q dimension
+  num_splits=0 means
+  it will be set by an internal heuristic. We're exposing num_splits mostly for
+  benchmarking. We will hard code it to 0 for now
+  */
+  constexpr int num_splits{0};
+  auto softmax_scale = std::pow(query.size(-1), -0.5);
+  //  CUDA code assumes that dout is contiguous
+  auto contiguous_grad_out = grad_out.contiguous();
+  auto contiguous_out = out.contiguous();
+  Tensor dq = at::empty_like(query);
+  Tensor dk = at::empty_like(key);
+  Tensor dv = at::empty_like(value);
+  //  The kernel computes irregadless we will drop for this functions return
+  Tensor grad_softmax;
+
+  std::tie(dq, dk, dv, grad_softmax) = fmha::mha_bwd(
+          contiguous_grad_out,
+          query,
+          key,
+          value,
+          contiguous_out,
+          logsumexp,
+          dq,
+          dk,
+          dv,
+          cumulative_sequence_length_q,
+          cumulative_sequence_length_k,
+          max_seqlen_batch_q,
+          max_seqlen_batch_k,
+          dropout_p,
+          softmax_scale,
+          false, /*zero_tensors = false for all calls here*/
+          is_causal,
+          num_splits,
+          (uint64_t)philox_seed, /*TODO: are these casts safe?*/
```

Comment:
Casts from int64_t to unint64_t. Some arithmetic is done on these numbers but this is a widening cast so should be safe.